### PR TITLE
LLM: add benchmark script for Max gpu and ipex fp16 gpu

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

add benchmark script for Max series GPU and ipex fp16 gpu.

### 1. Why the change?

Update oyr benchmark scripts for more hardwares (Max series GPU) and test cases (ipex fp16).

### 2. User API changes

No change.

### 3. Summary of the change 

- add benchmark script for Max series GPU (`run-max-gpu.sh`)
- add new benchmark api for ipex fp16 (`ipex_fp16_gpu`)

### 4. How to test?
- [x] Unit test
- [x] Local test

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/README.md [hunk 1]
```diff
@@ -27,8 +27,9 @@ test_api:
   - "native_int4"
   - "optimize_model"
   - "pytorch_autocast_bf16"
-  # - "transformer_int4_gpu"  # on arc
-  # - "optimize_model_gpu"  # on arc
+  # - "ipex_fp16_gpu" # on arc or pvc
+  # - "transformer_int4_gpu"  # on arc or pvc
+  # - "optimize_model_gpu"  # on arc or pvc
```
Comment (jason-dai) [inline python/llm/dev/benchmark/all-in-one/README.md:32]: on Intel GPU

File: python/llm/dev/benchmark/all-in-one/run-pvc.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+source /opt/intel/oneapi/setvars.sh
```
Comment (jason-dai) [inline python/llm/dev/benchmark/all-in-one/run-pvc.sh:1]: run max

File: python/llm/dev/benchmark/all-in-one/config.yaml [hunk 3]
```diff
@@ -13,5 +13,6 @@ test_api:
   - "native_int4"
   - "optimize_model"
   - "pytorch_autocast_bf16"
-  # - "transformer_int4_gpu"  # on arc
-  # - "optimize_model_gpu"  # on arc
\ No newline at end of file
+  # - "ipex_fp16_gpu" # on arc or pvc
+  # - "transformer_int4_gpu"  # on arc or pvc
+  # - "optimize_model_gpu"  # on arc or pvc
```
[the remark this answers is not in the document]
Comment (rnwang04) [reply python/llm/dev/benchmark/all-in-one/config.yaml]: Have revised to `on Intel GPU`

File: python/llm/dev/benchmark/all-in-one/README.md (hunk 1 above)
Comment (rnwang04) [reply python/llm/dev/benchmark/all-in-one/README.md]: Have revised to `on Intel GPU`

File: python/llm/dev/benchmark/all-in-one/run-pvc.sh (hunk 2 above)
Comment (jason-dai) [reply python/llm/dev/benchmark/all-in-one/run-pvc.sh]: maybe `run-max-gpu.sh` is better (as we also have Xeon Max)

Comment (rnwang04) [reply python/llm/dev/benchmark/all-in-one/run-pvc.sh]: Done

**Comment by jason-dai** [comment]
LGTM

**Review by plusbang — APPROVED** [review]

**Pull request merged**